api/src/main/java/org/openmrs/api/db/hibernate/HibernateCohortDAO.java

```diff
@@ -51,26 +53,42 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 	public Cohort getCohort(Integer id) throws DAOException {
 		return (Cohort) sessionFactory.getCurrentSession().get(Cohort.class, id);
 	}
-	
+
+	/**
+	 * @see org.openmrs.api.db.CohortDAO#getCohortsContainingPatientId(java.lang.Integer)
+	 */
+	@SuppressWarnings("unchecked")
+	public List<Cohort> getCohortsContainingPatientId(Integer patientId, Boolean voided) throws DAOException {
+		Disjunction orDisjunction = Restrictions.disjunction();
+		orDisjunction.add(Restrictions.isNull("m.endDate"));
+		orDisjunction.add(Restrictions.gt("m.endDate", new Date()));
+		return (List<Cohort>) sessionFactory.getCurrentSession().createCriteria(Cohort.class)
+				.add(Restrictions.eq("voided", voided))
+				.createAlias("members", "m")
+				.add(Restrictions.le("m.startDate", new Date()))
+				.add(orDisjunction)
+				.createAlias("m.patient", "p")
+				.add(Restrictions.eq("p.patientId", patientId))
```

Comment:
Can you please put your new comments below older ones and not above? Otherwise it's hard to follow them. 

The patient objects might be different but they have the same patientId so they are essentially the same from hibernate's perspective. 

You also need to make sure the test data is set up well, the only reason cohort is null for a membership is because it is actually not set in the DB or it isn't mapped correctly.